### PR TITLE
[Send funds] Add owner key first

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1839

Changes proposed in this pull request:
- Add owner key first screen

image

@gnosis/mobile-devs

### CODE REVIEW

**Comment by biafra23** [comment]
When I go through adding an owner, I am back at square one and have to restart the whole send funds flow. Also the imported key is not recognized and I have t go through the add key flow again. I would expect to be directed to the "select asset screen" after adding one key. 

The owner is recognized after going to the settings screen.

File: app/src/main/java/io/gnosis/safe/ui/safe/send_funds/AddOwnerFirstFragment.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import io.gnosis.safe.R
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.databinding.FragmentAddOwnerFirstBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.fragment.BaseViewBindingFragment
+import java.math.BigInteger
+
+class AddOwnerFirstFragment : BaseViewBindingFragment<FragmentAddOwnerFirstBinding>() {
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/safe/send_funds/AddOwnerFirstFragment.kt:15]: Do we need to track when AddOwnerFirstFragment was opened? Maybe not a question for this PR. But where would I put it?

File: data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt [hunk 2]
```diff
@@ -64,23 +64,56 @@ class SafeRepository(
...
+    }
+
+    suspend fun setActiveSafeSigningOwners(owners: List<Solidity.Address>) {
+        preferencesManager.prefs.edit {
+            putString(
+                ACTIVE_SAFE_SIGNING_OWNERS,
+                if(owners.isEmpty()) {
+                    null
+                } else {
+                    owners.joinToString(";") {
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt:104]: Should ";" be extarcted as for example `ADDRESS_SEPARATOR` constant?

**Review by biafra23 — APPROVED** [review]
LGTM 🚀

**Comment by biafra23** [comment]
Is it ok not to continue to the select asset screen after adding the owner keys has been added?

File: app/src/main/java/io/gnosis/safe/ui/safe/send_funds/AddOwnerFirstFragment.kt (hunk 1 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/safe/send_funds/AddOwnerFirstFragment.kt:15]: it is being tracked already. This fragment has screen id `screen_add_owner_to_send_assets`

File: data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt (hunk 2 above)
Comment (elgatovital) [reply data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt:104]: yes, as it is used in multiple places

**Pull request merged**